### PR TITLE
rgb565 support with arm neon optimization

### PR DESCRIPTION
**Pull request by sora-blue**
from_rgb565 supported currently

later on:
to_rgb565
color convert functions
test_max_pixel.cpp for from_rgb565

now confused about naming color convert functions, how about from_rgb565_2_gray?

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=3149) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=3149) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/ncnn?pullRequest=3149) it.</sub>

**Comment by nihui** [comment]
The conversion part is not efficient. You could refer the following articles for fast implementation.

https://community.arm.com/developer/ip-products/processors/b/processors-ip-blog/posts/coding-for-neon---part-4-shifting-left-and-right

https://developer.arm.com/documentation/den0018/a/NEON-Code-Examples-with-Optimization/Converting-color-depth/Converting-from-RGB565-to-RGB888
https://developer.arm.com/documentation/den0018/a/NEON-Code-Examples-with-Optimization/Converting-color-depth/Converting-from-RGB888-to-RGB565

**Comment by nihui** [comment]
For function name, rgb2gray_565 may be a good candidate

**Comment by BlueStar233** [comment]
saw that originally,  data of each channel is stored in float* with m.channel(int)
will i need to store raw rgb565 data in ncnn::Mat, instead of converting it into rgb888 before storing?

File: src/mat_pixel.cpp [hunk 1]
```diff
@@ -745,6 +886,20 @@ static void to_bgr2rgb(const Mat& m, unsigned char* rgb, int stride)
     }
 }
 
+static void to_bgr2rgb_565(const Mat& m, unsigned char* rgb565, int stride)
+{
+    int w = m.w;
+    int h = m.h;
+    unsigned char* rgb = new unsigned char[h * 1ll *stride];
```
Comment (BlueStar233) [inline src/mat_pixel.cpp:893]: it will be [w * h * 3]

**Pull request closed without merging**